post-processor/vsphere/post-processor.go

```diff
@@ -170,20 +178,24 @@ func (p *PostProcessor) PostProcess(ctx context.Context, ui packer.Ui, artifact
 	cmd.Stderr = errWriter
 
 	if err := cmd.Run(); err != nil {
-		err := fmt.Errorf("Error uploading virtual machine: %s\n%s\n", err, p.filterLog(errOut.String()))
+		err := fmt.Errorf("Error uploading virtual machine: %s\n%s\n", err, filterLog(errOut.String(), ovftool_uri))
 		return nil, false, false, err
 	}
 
-	ui.Message(p.filterLog(errOut.String()))
+	ui.Message(filterLog(errOut.String(), ovftool_uri))
 
 	artifact = NewArtifact(p.config.Datastore, p.config.VMFolder, p.config.VMName, artifact.Files())
 
 	return artifact, false, false, nil
 }
 
-func (p *PostProcessor) filterLog(s string) string {
-	password := escapeWithSpaces(p.config.Password)
-	return strings.Replace(s, password, "<password>", -1)
+func filterLog(s string, u *url.URL) string {
+	password, passwordSet := u.User.Password()
+	if passwordSet {
```

Comment:
```suggestion
	if passwordSet && password != "" {
```

Just in case the password is set but empty ? https://play.golang.org/p/pYecpKxnuVf